File: doc/.blinky/blinky.cpp
#include <cstdio>
#include <unistd.h>

#include <raspberrypi.h>

int main(void)
{
  puts("\nMuntsOS C++ LED Test\n");

  // Configure a GPIO output to drive an LED

  Interfaces::GPIO::Pin LED =
    new libsimpleio::GPIO::Pin_Class(RaspberryPi::GPIO26,
      Interfaces::GPIO::OUTPUT, false);

  // Flash the LED forever (until killed)

  puts("Press CONTROL-C to exit.\n");

  for (;;)
  {
    *LED = ! *LED;
    usleep(500000); // microseconds = 0.5 seconds
  }
}

File: doc/.blinky/blinky.java
import com.munts.interfaces.GPIO.*;
import com.munts.libsimpleio.objects.GPIO.*;
import com.munts.libsimpleio.platforms.RaspberryPi;

public class blinky
{
  public static void main(String args[])
    throws InterruptedException
  {
    System.out.println("\nMuntsOS Java LED Test\n");

    // Configure a GPIO output to drive an LED

    Builder b = new Builder(RaspberryPi.GPIO26);
    b.SetDirection(Direction.Output);
    Pin LED = b.Create();

    // Flash the LED forever (until killed)

    System.out.println("Press CONTROL-C to exit.\n");

    for (;;)
    {
      LED.write(!LED.read());
      Thread.sleep(500);
    }
  }
}
